ipfx/bin/generate_qc_input.py

```diff
@@ -6,46 +6,71 @@
     Generates an input JSON for QC module saved in the MODULE_IO_DIR
 """
 
-MODULE_IO_DIR = "../../tests/module_io"
+MODULE_IO_DIR = "../../tests/module_io/Ephys_Roi_Result_500844779"
 
-#output_json = os.path.join(MODULE_IO_DIR,'se_output_ivscc.json')
-#input_json = os.path.join(MODULE_IO_DIR,'se_input_ivscc.json')
+FEATURE_SUBSET = ["stimulus_units",
+               "stimulus_duration",
+               "sweep_number",
+               "vm_delta_mv",
+               "pre_noise_rms_mv",
+               "slow_noise_rms_mv",
+               "stimulus_scale_factor",
+               "post_noise_rms_mv",
+               "slow_vm_mv",
+               "stimulus_code",
+               "stimulus_name",
+               ]
 
-output_json = os.path.join(MODULE_IO_DIR,'se_output_patchseq.json')
-input_json = os.path.join(MODULE_IO_DIR,'se_input_patchseq.json')
 
 
-outd = ju.read(output_json)
-inpd = ju.read(input_json)
+def extract_sweep_features_subset(sweep_features, FEATURE_SUBSET):
 
-d = {}
+    sweep_features_subset = []
+    for sf in sweep_features:
+        sf_subset = {k: sf[k] for k in FEATURE_SUBSET}
+        sweep_features_subset.append(sf_subset)
 
```

Comment:
You could also just nest a dictionary comprehension in a list comprehension. It's pretty readable to me, but mileage may vary....

`sweep_features_subset = [{k: sf[k] for k in FEATURE_SUBSET} for sf in sweep_features]`